src/App.tsx

```diff
@@ -15,36 +17,92 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE = '',
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.NONE);
+  const [reversed, setReversed] = useState(false);
+
+  const getSortedGoods = () => {
+    const sortedGoods = [...goodsFromServer].sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.ALPHABET:
+          return good1.localeCompare(good2);
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+
+    if (reversed) {
+      sortedGoods.reverse();
+    }
+
+    return sortedGoods;
+  };
+
+  const visibleGoods = getSortedGoods();
+
+  const reset = () => {
+    setSortField(SortType.NONE);
+    setReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortField !== SortType.ALPHABET,
+          })}
+          onClick={() => setSortField(SortType.ALPHABET)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortField !== SortType.LENGTH,
+          })}
+          onClick={() => setSortField(SortType.LENGTH)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !reversed,
+          })}
+          onClick={() => setReversed(!reversed)}
```

Comment:
```suggestion
          onClick={() => setReversed(prevIsReversed => !prevIsReversed)}
```

https://dev.to/gakii/functional-state-update-in-react-42io